nixos/doc/manual/installation/installing-usb.section.md

```diff
@@ -1,31 +1,49 @@
-# Booting from a USB Drive {#sec-booting-from-usb}
+# Booting from a USB flash drive {#sec-booting-from-usb}
 
-For systems without CD drive, the NixOS live CD can be booted from a USB
-stick. You can use the `dd` utility to write the image:
-`dd if=path-to-image of=/dev/sdX`. Be careful about specifying the correct
-drive; you can use the `lsblk` command to get a list of block devices.
+The image has to be written verbatim to the USB flash drive for it to be
+bootable on UEFI and BIOS systems. Here are the recommended tools to do that.
 
-::: {.note}
-::: {.title}
-On macOS
+## Creating bootable USB flash drive with a graphical tool {#sec-booting-from-usb-graphical}
+
+Etcher is a popular and user-friendly tool. It works on Linux, Windows and macOS.
+
+Download it from [balena.io](https://www.balena.io/etcher/), start the program,
+select the downloaded NixOS ISO, then select the USB flash drive and flash it.
+
+::: {.warning}
+Etcher reports errors and usage statistics by default, which can be disabled in
+the settings. Some users might not like that.
 :::
 
+An alternative is [USBImager](https://gitlab.com/bztsrc/usbimager/#usbimager),
+which is very simple and does not connect to the internet. Download the version
+with write-only (wo) interface for your system. Start the program,
+select the image, select the USB flash drive and click "Write".
+
+## Creating bootable USB flash drive from Terminal on Linux {#sec-booting-from-usb-linux}
+
+Use the `dd` utility to write the image to the USB flash drive:
+
 ```ShellSession
-$ diskutil list
-[..]
-/dev/diskN (external, physical):
-   #:                       TYPE NAME                    SIZE       IDENTIFIER
-[..]
-$ diskutil unmountDisk diskN
-Unmount of all volumes on diskN was successful
-$ sudo dd if=nix.iso of=/dev/rdiskN bs=1M
+sudo dd if=<path-to-image> of=/dev/sdX bs=4M conv=fsync oflag=direct status=progress
 ```
 
-Using the \'raw\' `rdiskN` device instead of `diskN` completes in
+Be careful to replace `sdX` with the correct device! You can use the `lsblk`
+command to get a list of block devices and their sizes.
+
+## Creating bootable USB flash drive from Terminal on macOS {#sec-booting-from-usb-macos}
+
+Use the `dd` utility to write the image to the USB flash drive. Unmount the disk before.
```

Comment:
That last sentence feels *off*. It should anyway probably be written before the instruction to use `dd`. Something like *make sure the drive is unmounted.*, or any other preferred formulation.

This is probably needed in the Linux instructions too.

Reply:
to test if unmounting is needed, i have created 3 fat partitions on a usb drive, mounted them in my file browser (elementary Files, which use GVFS), created some folders and copied some files, then used dd to write the image, the partitions where still mounted and i was able to copy files, but unmounting takes very long... then i see only one partition (the first), i can mount it again, but all files copied after the dd are messed up

![Screenshot from 2022-05-29 18-56-43](https://user-images.githubusercontent.com/91113/170882256-b5acd2f3-7f2d-4e52-87d9-0bf92bf024ff.png)

GNOME Files still shows the old partitions before the dd

when disconnecting the drive and plugging it back in, it looks correct. so the dd should work without unmounting, but your programs will be confused about the mounted partitions and you probably loose data copied there into the void

i add instructions to unmount